kivy/data/style.kv

```diff
@@ -490,14 +490,14 @@
 
 # ModalView widget
 <ModalView>:
-    canvas:
+    canvas.before:
```

Comment:
Is there a particular reason to use `canvas.before`? I think `canvas` is probably a better choice.

Reply:
I was having trouble with it since the usual way to change the background image via canvas is using canvas.before, and when you try to use a canvas.before on a Popup (which is a child class of ModalView), it is completely overlayed by this, and the only way to effectively nullify this is to use a transparent png as the background image.

Reply:
As @matham mentioned on discord, the canvas is itself ordered background to foreground, the point of canvas.before and canvas.after is to bypass that ordering when adding instructions in different places (in which case managing the single flattened canvas list is inconvenient). When you make a widget like this, you have full control over the ordering, so it's usually best to do everything on the normal canvas.

This is especially important for Kivy's own widgets, because these are building blocks that many apps will use. They generally want to avoid using the canvas.before and canvas.after specifically because that makes it harder for users to do their own drawing before and after the Kivy widget. Also note that 'drawing' encompasses more than displaying shapes, but also context instructions, see below.

Your change here is well motivated, but I think it's still the worse choice - if anything, the fact that the ModalView draws an opaque background image by default may be the bad part of the api. As I would see it, this change breaks the normal convention: you've specifically made canvas.before draw *after* the widget, which is the opposite of its normal behaviour. I understand why it may be initially surprising that drawing a background before everything else means it's behind the image that the widget displays...but isn't that what you should expect? That's why it's called canvas.before!

There may be other practical issues. For instance, users would normally expect that using a Rotate or Translate instruction on canvas.before would affect the widget, but this would not work here.

For these reasons, I think using canvas is the most consistent and preferable option.

Reply:
Also, setting the `background_color` to have 0 opacity should be a more convenient alternative to using a transparent png.

Reply:
ok, you are right about the background_color, i was thinking about adding this with the way it used to behave (i had to do the hack of making the background transparent on my app to get it to work properly).
What do you think about making the opacity fadeout thing a canvas.before, and the background image itself canvas?  Not asking for any usability reasons, just does that seem to make more sense?

Reply:
My initial reaction is that even the fadeout should probably be on the canvas rather than canvas.before, simply because I don't see a major reason not to do it, but if you have a good reason for it then that could be fine.

Reply:
Honestly, I dont have a good reason for putting it on canvas.before, just thought it made sense so I would suggest it, either way is fine by me.

Reply:
Changed it back to just use canvas.